src/olympia/addons/indexers.py

```diff
@@ -388,3 +393,120 @@ def extract_document(cls, obj):
                     cls.extract_field_raw_translations(preview, 'caption'))
 
         return data
+
+
+# addons index settings.
+INDEX_SETTINGS = {
+    'analysis': {
+        'analyzer': {
+            'standardPlusWordDelimiter': {
+                # This analyzer tries to split the text into words by using
+                # various methods. It also lowercases them and make sure each
+                # token is only returned once.
+                # Only use for short things with extremely meaningful content
+                # like add-on name - it makes too many modifications to be
+                # useful for things like descriptions, for instance.
+                'tokenizer': 'standard',
+                'filter': [
+                    'standard', 'wordDelim', 'lowercase', 'stop', 'dict',
+                    'unique'
+                ]
+            }
+        },
+        'normalizer': {
+            'lowercase_keyword_normalizer': {
+                'type': 'custom',
+                'filter': ['lowercase'],
+            },
+        },
+        'filter': {
+            'wordDelim': {
+                # This filter is useful for add-on names that have multiple
+                # words sticked together in a way that is easy to recognize,
+                # like FooBar, which should be indexed as FooBar and Foo Bar.
+                # (preserve_original: True makes us index both the original
+                # and the split version.)
+                'type': 'word_delimiter',
+                'preserve_original': True
+            },
+            'dict': {
+                # This filter is also useful for add-on names that have
+                # multiple words sticked together, but without a pattern that
+                # we can automatically recognize. To deal with those, we use
+                # a small dictionary of common words. It allows us to index
+                # "awesometabpassword"  as "awesome tab password", helping
+                # users looking for "tab password" find that add-on.
+                'type': 'dictionary_decompounder',
+                'word_list': [
```

Comment:
these are all English words, right?  How does it work with non-English?

Reply:
It doesn't, this only really helps english - but we also decompose words automatically according to various criteria (CamelCase, any non-letter, etc)